back/src/types/user.dto.ts

```diff
@@ -1,5 +1,12 @@
-import { IsNumber, IsString, IsNotEmpty } from 'class-validator';
+import { IsNumber, IsString, IsNotEmpty, Length, Matches } from 'class-validator';
 
+export class UsernameDto {
+	@IsNotEmpty({ message: 'Username is required' })
+	@IsString({ message: 'Username must be a string' })
+	@Length(3, 32, { message: 'Username must be between 3 and 32 characters' })
+	@Matches(/^[a-zA-Z0-9_]+$/, { message: 'Username must only contain alphanumeric characters and underscores' })
+	username: string;
```

Comment:
You need to allow ``-`` as well.
Some 42 usernames contain hyphens.

Reply:
Done!